### PR TITLE
[1LP][RFR]New tag expression tests

### PR DESCRIPTION
**Pull request by rlbabyuk**
Purpose or Intent
=================
{{pytest: cfme/tests/configure/test_access_control.py cfme/tests/cloud_infra_common/test_tag_visibility.py  -k 'test_group_crud_with_tag or test_tag_expression' --use-provider rhv42 --use-provider ec2west --use-provider vsphere6-nested }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0f2db11130966e0e8ac52441a26a2f427c7ba103:

`cfme/tests/cloud_infra_common/test_tag_visibility.py`:
- :red_circle: Line 8:1: **F401** *'cfme.infrastructure.virtual_machines.InfraVm' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/configure/test_access_control.py [hunk 1]
```diff
@@ -85,6 +85,24 @@ def _check_item_visibility(item, user_restricted):
...
+        tag_for_update = ([category.display_name, tag.display_name], False)
+    else:
+        if appliance.version > '5.9':
+            if BZ(1579867, forced_streams=['5.9']).blocks:
+                tag_for_create = 'fill_tag(My Company Tags : Cost Center, Cost Center 001)'
+            else:
+                tag_for_create = 'fill_tag(My Company Tags : {})'.format(category.display_name)
+            tag_for_update = 'delete_whole_expression'
+        else:
+            pytest.skip("Tag expression is suported from 5.9 version")
```
Comment (mshriver) [inline cfme/tests/configure/test_access_control.py]: This should be an uncollect marker applied to the test case, not a skip, which shows up as a failure in 58z.

File: cfme/tests/configure/test_access_control.py [hunk 2]
```diff
@@ -85,6 +85,24 @@ def _check_item_visibility(item, user_restricted):
...
 
 
+@pytest.fixture(params=['tag', 'tag_expression'])
+def tag_value(appliance, category, tag, request):
+    tag_type = request.param
+    if tag_type == 'tag':
+        tag_for_create = ([category.display_name, tag.display_name], True)
+        tag_for_update = ([category.display_name, tag.display_name], False)
+    else:
+        if appliance.version > '5.9':
```
Comment (mshriver) [inline cfme/tests/configure/test_access_control.py]: This should be `if appliance.version >= '5.9'` yes? or was it only active in a Z-stream?

File: cfme/tests/cloud_infra_common/test_tag_visibility.py [hunk 3]
```diff
@@ -25,3 +31,216 @@ def tagged_vm(tag, has_no_providers_modscope, setup_provider_modscope, provider)
...
+    category = Category(name='service_level',
+                        display_name='Service Level')
+    tag = Tag(name='silver',
+              display_name='Silver',
+              category=category)
+    return tag
+
+
+@pytest.fixture(scope='module')
+def vms_for_tagging(provider, request, appliance):
```
Comment (quarckster) [inline cfme/tests/cloud_infra_common/test_tag_visibility.py]: **OPTIONAL**
`request` is not used here.

File: cfme/tests/cloud_infra_common/test_tag_visibility.py [hunk 4]
```diff
@@ -25,3 +31,216 @@ def tagged_vm(tag, has_no_providers_modscope, setup_provider_modscope, provider)
...
+        with user_restricted:
+            view = navigate_to(appliance.server, 'LoggedIn')
+
+            orig_group = view.current_groupname
+            if group.description != orig_group:
+                view.change_group(group.description)
+            try:
+                navigate_to(vm, 'Details')
+                actual_visibility = True
+            except Exception:
```
Comment (quarckster) [inline cfme/tests/cloud_infra_common/test_tag_visibility.py]: **REQUIRED**
Please be more specific, what kind of exception should raise here?

**Comment by mshriver** [comment]
restarted PRT with rhv42 and ec2west specified as providers, vsphere65-nested is failing to setup ATM

**Comment by mshriver** [comment]
Changes look fine, trying to get some PRT results collected, skipping on provider.

**Pull request merged**